openedx/frontend-app-course-authoring
# feat: Create collection Modal [FC-0062]

## Description

Adds a functional Modal to create a library collection.

![image](https://github.com/user-attachments/assets/5e6cc3d4-f677-467a-8312-bb6051c192fc)

## Supporting information

- Github link: https://github.com/openedx/frontend-app-course-authoring/issues/1100
- Internal ticket: [FAL-3788](https://tasks.opencraft.com/browse/FAL-3788)
- Depends on: https://github.com/open-craft/edx-platform/pull/683

## Testing instructions

- Use this branch of `edx-platform`: https://github.com/open-craft/edx-platform/pull/683
- Go to studio and create a library
- In the library home click on the `New` button
- Click the' Collection' button on the `Add content` sidebar.
- Verify the design of the modal.
- Click on `Create` with all empty. Verify the validation error.
- Add a name and click on `Create`. Verify the success.
- Run `tutor dev run cms ./manage.py cms reindex_studio --experimental` and refresh the library home page to see the newly created collection.
- Return to the create modal and add the same name of the newly created collection. Click on `Create` and verify the validation error.
